### PR TITLE
ci: fix coverage + minor touch ups

### PR DESCRIPTION
**Pull request by Saransh-cpp**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -3,13 +3,16 @@ name: BPX-tests
 on:
   workflow_dispatch:
   pull_request:
+  push:
+    branches:
+      - main
```
Comment (Saransh-cpp) [inline .github/workflows/test.yml:8]: CI should always run on the default branch. Things like codecov will not work if there is no coverage report uploaded from the default branch. See how [codecov says "no reports found"](https://app.codecov.io/gh/FaradayInstitution/BPX)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -3,13 +3,16 @@ name: BPX-tests
...
 
 jobs:
   build:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```
Comment (Saransh-cpp) [inline .github/workflows/test.yml]: If BPX is available on 3.12, why not test on 3.12

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -3,13 +3,16 @@ name: BPX-tests
...
+    branches:
+      - main
 
 jobs:
   build:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
```
Comment (Saransh-cpp) [inline .github/workflows/test.yml:15]: If BPX is available on 3.12 and 3.12, why not test on them?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -21,8 +24,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install hatch
-          pip install .
+          pip install hatch "coverage[toml]"
```
Comment (Saransh-cpp) [inline .github/workflows/test.yml:27]: `pytest-cov` requires `coverage.py` to be installed globally. See the silently failing logs in the previous PRs ([silent failure](https://github.com/FaradayInstitution/BPX/actions/runs/12437661489/job/34728010041) in #84; [silent failure](https://github.com/FaradayInstitution/BPX/actions/runs/12436303784/job/34723880823) in #83).

Also see how codecov finally posts a coverage comment on this PR.

File: README.md [hunk 5]
```diff
@@ -12,10 +12,17 @@ To see how to use BPX with [PyBaMM](https://www.pybamm.org/), check out the [BPX
...
+
 ```bash
 pip install bpx
 ```
 
+or conda
+
+```bash
+conda install -c conda-forge bpx
+```
```
Comment (Saransh-cpp) [inline README.md:24]: BPX is available on conda-forge now 😉

**Comment by Saransh-cpp** [comment]
Cannot request a review here as I am not a maintainer.

cc: @rtimms @kratman

**Review by rtimms — APPROVED** [review]

**Pull request merged**